### ISSUE TITLE
Unable to delete VPC connector

### ISSUE THREAD

**Issue by emilhdiaz** [aws/apprunner-roadmap#161]
**Community Note**
* Please vote on this issue by adding a 👍 reaction to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do * not help prioritize the request
If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
What do you want us to build?
Attempted to delete an unused VPC Connector through the AWS CLI and the error message received is unhelpful
```
PAGER= 
aws apprunner list-services
{
    "ServiceSummaryList": []
}

aws apprunner delete-vpc-connector --vpc-connector-arn <REDACTED>

An error occurred (InvalidRequestException) when calling the DeleteVpcConnector operation: Provided VPC connector arn is being used by other services. ARN:  <REDACTED>
```
Details as to which "other services" are still using the connector would help users understand how to resolve the issue. 

**Describe alternatives you've considered**
Deleting through the AWS Console equally produces an unhelpful error message: 

```
Failed to delete VPC Connector
Provided VPC connector arn is being used by other services. ARN: <REDACTED>
```

**Additional context**
Anything else we should know? 
To clarify, this issue seems to occur after deleting an AppRunner service that previously have this VPC connector attached. 

**Attachments**
If you think you might have additional information that you'd like to include via an attachment, please do - we'll take a look. (Remember to remove any personally-identifiable information.)
N/A

**Comment by hariohmprasath** [comment]
Hi @emilhdiaz,
I work for App Runner service team and we will be happy to help you with this issue. Can you please provide the VPC connector ARN so we can help you out on this? Thanks

**Comment by emilhdiaz** [comment]
Hi @hariohmprasath, 

Thanks for the help! Sure thing there are two of them: 

arn:aws:apprunner:us-east-2:265941575127:vpcconnector/dev01-valtio-api-vpc-connector/1/9648e87f03da4f1da59de08117141785

arn:aws:apprunner:us-east-2:265941575127:vpcconnector/dev01-valtio-api-egress-connection/1/9ffce5c538c54e18b855fb8d031f59c4

**Comment by hariohmprasath** [comment]
Hi @emilhdiaz,
Sorry for the inconvenience caused, we have resolved the issue on our end, can you please retry now? Thanks

**Comment by emilhdiaz** [comment]
@hariohmprasath I can confirm that I was now able to delete those two VPC connectors. Thank you very much! 

Out of curiosity, was the resolution here specific to these two VPC connectors or will it also prevent this from occurring in the future? Reason I ask is because I am trying to Terraform these resources and thus resources getting stuck and not being able to delete are problematic for maintaining Terraform state.

**Comment by academiaresf** [comment]
hi, i have exactly the same problem. Terraform tries to recreate the connector and the CI is stucked.

**Comment by kyle-eastcode** [comment]
Hello @hariohmprasath 
I seem to be having the same issue with one of my app runner services, looks like I can't delete the app runner service with the following error in the aws console:
You cannot delete a Service while it has VpcIngressConnections with a status of FAILED_CREATION.

**Comment by SebastienMelki** [comment]
I am having the same issue here. @hariohmprasath hariohmprasath  even though a vpc-connector is disconnected from a particular service in apprunner, it is still undeletable. If I delete the service (that is no longer connected), then it becomes deletable. This is inconvenient.

Proof in screenshots

![image](https://github.com/user-attachments/assets/18991c5a-9844-485f-81f4-313084c17429)
![image](https://github.com/user-attachments/assets/b3dcb441-7d47-4187-b42f-082078ac8c06)

**Comment by alin-plamadeala** [comment]
same issue here. have to delete app runner service before i can delete the vpc connector

**Comment by qwatts-dev** [comment]
Any updates here? Will I need to delete my app runner service before I can delete a vpc connector?

**Comment by kyle-eastcode** [comment]
Update: I got it resolved, you have to delete the vpc endpoint via the AWS cli with a specific way you call the cli, the docs might have been updated since I posted this so hopefully following the docs for deleting a vpc endpoint via cli will help you resolve it

**Comment by kyle-eastcode** [comment]
@qwatts-dev yes deleting the service might work, or if you can't delete it, you might have delete the vpc endpoint manually via the AWS cli
